### ISSUE TITLE
User Roles?

### ISSUE THREAD

**Issue by SimonLab** [SafeLives/SafeLives#97]
@JoshTaylor12 could you confirm the following list of roles, the users linked to these roles will be allowed to login to the application. This will help me to determine the permissions and scope for each type of user (thanks @harrygfox  for this list):

- SafeLives Admin
- Commissioner (operates between services)
- Service Manager
- Service Worker
  - Insights Champion (main point of contact for a service)
  - IDVA (Independent Domestic Violence Advisor)
- Client (abuse victim, not allow to login into the application)

- I'm not sure about the type IDVA, is it the same as Service Worker, a sub type of Service Worker or another type?
- I'm not sure also about Commissioner, does this type as a specific role description or is it also the same as Service Worker?
- Is Service Manager a sub type of Service Worker?

Are there any other type of user that will need access to the application?
@JoshTaylor12 @iteles @harrygfox any thoughts on this list?

**Comment by JoshTaylor12** [comment]
@SimonLab So first I'll answer the questions and then below that would be confirmation of the list.

Idva would be a type of service worker. There are also other roles that I will list out below. 

The Insights champion will also be in one of these roles. The Insights champion could also be the service manager. So for this reason I would have it as a tag that you can assign to anyone, rather than a separate role.

Commissioner would be separate to the service workers, and their permissions would be severely restricted.

Service manager would be best placed separate to the service workers

**Comment by JoshTaylor12** [comment]
@SimonLab this would be the list of roles:

- SafeLives admin
- commissioner
- service manager
- case worker

There would then be many subsets to the case worker role. This gets complicated because the service might call their workers something that is unique to them. So ideally this would need to be a list that can be added to. To add to this, we will group some roles together to create our Insights datasets. So for example, we have a service that has Marac Idvas and SDVC Idvas, but the data from both roles would go into the general Idva dataset. 

For this reason, i think we would need two tags. One for role, and one for the dataset that the role belongs to.

The current list of case worker sub types are:

Role	                                        Dataset
IDVA	                                Idva
Outreach worker	                Outreach
Male Outreach	                Outreach
Women's safety worker	        Women's safety worker
Refuge Worker	               Refuge
Advice worker   	               Advice/Helpline
SDVC IDVA	                       Idva
MARAC IDVA	                       Idva
YP Worker	                       YP Worker
Helpline	                               Advice/Helpline
Isva	                                       Isva
Boy and Young Men Worker  Outreach
BME Worker	                       BME worker
Refuge childrens worker	       Children's
Children's IDVA	               Children's
Children's Counsellor	       Children's
Children's Outreach	       Children's
Health IDVA	                       Health
IRIS IDVA	                              Health
Maternity IDVA	              Health
A&E  IDVA	                      Health
Floating Support	             Outreach
Support worker	             Outreach
Idaa	                                      Idaa

**Comment by SimonLab** [comment]
Thanks for the details @JoshTaylor12 !
Just to recap:
The sub types of case worker is a way to identify easily the different type users in a service but otherwise these sub types will have the same permissions (list of features available to this role) defined by the case worker type or is it possible each sub types could have different restricted permissions?

These sub types of case worker can also be grouped to create specific dataset. You've mentioned the "Idva dataset", do you have other specific dataset created from these sub types?
Sorry we might have talked about the datasets previously but could you precise them just to be sure I don't miss anything. From my understanding a dataset is the analysed data from the forms submitted by case workers. Is a specific dataset the same as a report generated for a specific role?
![report-role](https://cloud.githubusercontent.com/assets/6057298/21858173/124b10b6-d820-11e6-93f3-729d6354c6b0.png)

**Comment by JoshTaylor12** [comment]
@SimonLab that's correct, these sub types would all have the same permission levels.

Sorry the previous list didnt come out well when i wrote it in so I have attached a file showing which dataset each role should feed into. It would also be important that SafeLives admin users were able to add to this list.

I think we did talk about datasets but probably didnt make it clear that more than one role would feed into them. So the datasets are made up of similar roles. 

We would still require the function to report by single role (as in the wireframe above) so dataset should probably sit as a separate category underneath.

**Comment by SimonLab** [comment]
Thanks for the explanation @JoshTaylor12. I wasn't paying attention to the details and didn't see the two column on your previous issue for Role and dataset.

I have another question concerning the Service manager role. Does this role can create new case worker role or is the creation of new user only allowed by the Safe Lives Admins?

**Comment by JoshTaylor12** [comment]
Yes it would be good if both Service Manager and SafeLives Admin were able to create new caseworkers
